utils/mgmt_system/google.py

```diff
@@ -0,0 +1,269 @@
+# coding: utf-8
+"""Backend management system classes
+
+Used to communicate with providers without using CFME facilities
+"""
+
+import time
+from oauth2client.client import flow_from_clientsecrets
+from oauth2client.file import Storage
+from oauth2client.tools import run
+from googleapiclient.discovery import build
+from utils.mgmt_system.base import MgmtSystemAPIBase, VMInfo
+from utils.mgmt_system.exceptions import VMInstanceNotFound
+from utils.log import logger
+from utils.wait import wait_for
+
+
+class GoogleCloudSystem (MgmtSystemAPIBase):
+    """
+    Client to Google Cloud Platform API
+
+    """
+
+    states = {
+        'running': ('RUNNING',),
+        'stopped': ('TERMINATED',),
+        'starting': ('STAGING'),
+        'stopping': ('STOPPING'),
+    }
+
+    def __init__(self, **kwards):
+        self._project = kwards['project']
+        self._zone = kwards['zone']
+
+        # Default name of the instance name from config file
+        self._instance_name = kwards['default_instance_name']
+
+        # Perform OAuth 2.0 authorization.
+        # based on OAuth 2.0 client IDs credentials from client_secretes file
+        if kwards['client_secrets'] and kwards['scope'] and kwards['oauth2_storage']:
+            flow = flow_from_clientsecrets(kwards['client_secrets'], scope=kwards['scope'])
+            storage = Storage(kwards['oauth2_storage'])
+            self._credentials = storage.get()
+
+        if self._credentials is None or self._credentials.invalid:
+            self._credentials = run(flow, storage)
+
+        if self._credentials is None or self._credentials.invalid:
+            raise Exception("Incorrect credentials for Google Cloud System")
+
+        self._compute = build('compute', 'v1', credentials=self._credentials)
+
+    def _get_all_instances(self):
+        return self._compute.instances().list(project=self._project, zone=self._zone).execute()
+
+    def list_vm(self):
+        result = self._get_all_instances()
+        return result['items']
+
+    def _find_instance_by_name(self, instance_name):
+        instance = self._compute.instances().get(project=self._project,
+                                                 zone=self._zone,
+                                                 instance=instance_name).execute()
+        if instance:
+            return instance
+        else:
+            raise VMInstanceNotFound(instance_name)
+
+    # didn't use the utils.wait -> wait_for because native way for cheking the status
+    # is more safety and we can get additional information about error
```

Comment:
This should be incorporated into a wait_for, with wait_for we get enhanced logging and a timeout value, as we need to avoid scenarios for code to sit in a forever loop.